addons/promote/common/src/main/java/org/commonjava/indy/promote/validate/PromotionValidationTools.java

```diff
@@ -561,6 +563,66 @@ public Transfer getTransfer( final StoreResource resource )
         runParallelAndWait( entries, closure, logger );
     }
 
+    public <T> void paralleledInBatch( Collection<T> collection, int batchSize, Closure closure )
+    {
+        logger.trace( "Exe parallel on collection {} with closure {} in batch {}", collection, closure, batchSize );
+        Collection<Collection<T>> batches = batch( collection, batchSize );
+        runParallelInBatchAndWait( batches, closure, logger );
+    }
+
+    public <T> void paralleledInBatch( T[] array, int batchSize, Closure closure )
+    {
+        logger.trace( "Exe parallel on array {} with closure {} in batch {}", array, closure, batchSize );
+        Collection<Collection<T>> batches = batch( Arrays.asList( array ), batchSize );
+        runParallelInBatchAndWait( batches, closure, logger );
+    }
+
+    public <K, V> void paralleledInBatch( Map<K, V> map, int batchSize, Closure closure )
+    {
+        Set<Map.Entry<K, V>> entries = map.entrySet();
+        logger.trace( "Exe parallel on map {} with closure {} in batch {}", entries, closure, batchSize );
+        runParallelAndWait( entries, closure, logger );
```

Comment:
Is this a typo? Why this ParallelAndWait(or inBatch) executed twice?

Reply:
Good catch! This should be runParallelInBatchAndWait.

Reply:
Sorry. This should be removed :)